native/src/modules/categories/components/CategoryListContent.js

```diff
@@ -0,0 +1,181 @@
+    const { language, navigation, navigateToInternalLink } = this.props
+    onLinkPress(url, navigation, language, navigateToInternalLink)
+  }
+
+  alterResources = (node: HTMLNode) => {
+    const { cacheDictionary } = this.props
+    if (node.attribs) {
+      if (node.attribs.href) {
+        console.debug(`Found href: ${decodeURI(node.attribs.href)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.href)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.href} with ${newResource}`)
+          node.attribs = {
+            ...(node.attribs || {}),
+            href: newResource
+          }
+          return node
+        }
+      } else if (node.attribs.src) {
+        console.debug(`Found src: ${decodeURI(node.attribs.src)}`)
+        const newResource = cacheDictionary[decodeURI(node.attribs.src)]
+        if (newResource) {
+          console.debug(`Replaced ${node.attribs.src} with ${newResource}`)
+          node.attribs = {
+            ...(node.attribs || {}),
+            src: newResource
+          }
+          return node
+        }
+      }
+    }
+  }
+
+  // see https://github.com/archriss/react-native-render-html/issues/286
+  renderBulletLists: RendererFunction = (htmlAttribs, children, convertedCSSStyles, passProps) => {
+    const { language, theme } = this.props
+    const style = _constructStyles({
+      tagName: 'ul',
+      htmlAttribs,
+      passProps,
+      styleSet: 'VIEW'
+    })
+    const { allowFontScaling, rawChildren, nodeIndex, key, baseFontStyle, listsPrefixesRenderers } = passProps
+    const baseFontSize = baseFontStyle.fontSize
+    children = children && children.map((child, index) => {
+      const rawChild = rawChildren[index]
+      let prefix = false
+      const rendererArgs = [
+        htmlAttribs,
+        children,
+        convertedCSSStyles,
+        {
+          ...passProps,
+          index
+        }
+      ]
+      const textDistanceToBullet = 10
+      const rtlListIndent = 20
+      const bulletSizeRelativeToFont = 2.8
+      const bulletAlignmentRelativeToFont = 2
+      if (rawChild) {
+        if (rawChild.parentTag === 'ul' && rawChild.tagName === 'li') {
+          prefix = listsPrefixesRenderers && listsPrefixesRenderers.ul ? listsPrefixesRenderers.ul(...rendererArgs) : (
+            <View style={{
+              width: baseFontSize / bulletSizeRelativeToFont,
+              height: baseFontSize / bulletSizeRelativeToFont,
+              borderRadius: baseFontSize / bulletSizeRelativeToFont,
+              marginRight: RTL_LANGUAGES.includes(language) ? rtlListIndent : textDistanceToBullet,
+              marginLeft: RTL_LANGUAGES.includes(language) ? textDistanceToBullet : 0,
+              marginTop: baseFontSize / bulletAlignmentRelativeToFont,
+              backgroundColor: theme.colors.textColor
+            }} />
+          )
+        } else if (rawChild.parentTag === 'ol' && rawChild.tagName === 'li') {
```

Comment:
There are some magic values.

Reply:
you mean ul, ol and li? do you think we need to actually extract them? I think they speak for themselves